api/graphql/mutations/project.py

```diff
@@ -34,3 +52,108 @@ async def add_comment(
             entity=CommentEntityType.project, entity_id=id, content=content
         )
         return GraphQLComment.from_internal(result)
+
+    @strawberry.mutation
+    async def create_project(
+        self,
+        name: str,
+        dataset: str,
+        create_test_project: bool,
+        info: Info,
+    ) -> int:
+        """
+        Create a new project
+        """
+        connection = info.context['connection']
+        ptable = ProjectPermissionsTable(connection)
+
+        await ptable.check_project_creator_permissions(author=connection.author)
+
+        pid = await ptable.create_project(
+            project_name=name,
+            dataset_name=dataset,
+            author=connection.author,
+        )
+
+        if create_test_project:
+            await ptable.create_project(
+                project_name=name + '-test',
+                dataset_name=dataset,
+                author=connection.author,
+            )
+
+        return pid
+
+    @strawberry.mutation
+    async def update_project(
+        self,
+        project: str,
+        project_update_model: JSON,
+        info: Info,
+    ) -> JSON:
+        """Update a project by project name"""
+        connection = info.context['connection']
+        connection.check_access({ProjectMemberRole.project_admin})
+
+        ptable = ProjectPermissionsTable(connection)
+        await ptable.update_project(
+            project_name=project,
+            update=project_update_model,  # type: ignore [arg-type]
+            author=connection.author,  # type: ignore [arg-type]
+        )
+
+        return JSON({'success': True})
+
+    @strawberry.mutation
+    async def delete_project_data(
+        self,
+        delete_project: bool,
+        info: Info,
+    ) -> JSON:
+        """
+        Delete all data in a project by project name.
+        """
+        connection = info.context['connection']
+        connection.check_access({ProjectMemberRole.project_admin})
+
+        ptable = ProjectPermissionsTable(connection)
+        assert connection.project
```

Comment:
There won't be a project on the connection

Reply:
Fixed!